client/me/billing-history/upcoming-charges-table.jsx

```diff
@@ -1,62 +1,63 @@
 /**
  * External dependencies
  */
-import React, { PropTypes } from 'react';
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
 
 /**
  * Internal Dependencies
  */
-var	purchasesPaths = require( 'me/purchases/paths' ),
-	TransactionsTable = require( './transactions-table' );
+import purchasesPaths from 'me/purchases/paths';
+import TransactionsTable from './transactions-table';
+import { getSitesForUpcomingTransactions } from 'state/selectors';
+
+class UpcomingChargesTable extends Component {
+	static propTypes = {
+		// Computed props
+		sites: PropTypes.object.isRequired,
+	}
 
-module.exports = React.createClass( {
+	renderTransaction = ( transaction ) => {
+		const { translate } = this.props;
+		const site = this.props.sites[ Number( transaction.blog_id ) ];
 
-	displayName: 'UpcomingChargesTable',
+		if ( ! site ) {
+			return null;
```

Comment:
Clearing browser state, then visiting http://calypso.localhost:3000/me/purchases/billing now causes each transaction to be missing the `Manage Purchase` link. 

It looks like we want the full site object here just for a `site.slug` and transaction id. Can we update `getSitesForUpcomingTransactions` to just use ids, and use that in place of the site slug? We could maybe then have the purchase route redirect to the correct slug since we'd likely need to fetch a site in that case.